src/components/Avatar.svelte

```diff
@@ -4,29 +4,44 @@
     import type { NDKUser, NDKUserProfile } from "@nostr-dev-kit/ndk";
     import { currentIdentity } from "../stores/identities";
 
-    export let pubkey: string;
-    export let pxSize: number = 32;
+    interface Props {
+        pubkey: string;
+        pxSize?: number;
+        showRing?: boolean;
+    }
 
-    let user: NDKUser;
-    let profile: NDKUserProfile | null = null;
-    
-    onMount(async () => {
-        user = $ndk.getUser({ pubkey });
-        profile = await user.fetchProfile();
-    });
+    let { pubkey, pxSize = 32, showRing = false }: Props = $props();
```

Comment:
I don't understand why this is an improvement over the old `export let` stuff. What's the difference?

Reply:
The props like this are automatically reactive and deeply reactive so you can pass objects through if you want as well as simple values. It's the combination of that plus the derived values below that means loads of stuff just stays in sync.

Reply:
Interesting, I don't feel like I ever had that problem. Is it because $state variables proxy array methods like `push`? I tried using push the other day but it didn't trigger reactivity.

Reply:
I feel like fetching profile data especially has always been problematic for me. Might just be the way I'm using things in variables instead of const functions like you prefer. The state stuff is supposed to track changes to arrays, objects, etc. now. https://youtu.be/tErKyuUTzsM?si=1sVVoAmqPRrjqrss&t=611